marshmallow/utils.py

```diff
@@ -402,3 +402,16 @@ def get_func_args(func):
 
 def if_none(value, default):
     return value if value is not None else default
+
+
+def iterable_to_string(values):
+    """Format an iterable as a human readable string suitable for a sentence.
+    """
+    values = list(values)
+    if len(values) > 2:
+        return '{}, and {}'.format(', '.join(values[:-1]), values[-1])
+    if len(values) == 2:
+        return ' and '.join(values)
+    if len(values) == 1:
+        return values[0]
+    return 'None'
```

Comment:
Would it make more sense to raise an error instead of return `'None'` here? We *should*n't ever hit this, but it would be incorrect to get `'None is not a valid field for MySchema'`.

Reply:
As a generic utility it seemed like a logical representation of an empty list. It is currently impossible to hit this line from `Schema._update_fields()`.

Maybe we don't need to format the message as a single string though. Developers seem to be just fine with lists and dictionaries as the "message" value for validation exceptions. Any preference for the message structure?

Reply:
> Maybe we don't need to format the message as a single string though. Developers seem to be just fine with lists and dictionaries as the "message" value for validation exceptions. Any preference for the message structure?

Yep. I wouldn't bother creating a string.

```python
    if invalid_fields:
        raise AttributeError("Invalid fields for {0}: {1}".format(self.__class__.__name__, invalid_fields))
```

Validation errors may be exposed to end users (like when using webargs) or even parsed by API clients, so they should be consistent, but this is just meant to be seen by the developer after a typo, so don't overthink it.